lighthouse-core/lib/traces/tracing-processor.js

```diff
@@ -237,6 +237,20 @@ class TraceProcessor {
       }
     }
 
+    // Support the case where everything else fails.
+    const navStartEvt = events.find(e => Boolean(e.name === 'navigationStart' && e.args &&
+      e.args.data && e.args.data.isLoadingMainFrame && e.args.data.documentLoaderURL));
+    if (navStartEvt && navStartEvt.args && navStartEvt.args.data) {
+      const frameId = navStartEvt.args.frame;
+      if (frameId) {
+        return {
+          pid: navStartEvt.pid,
+          tid: navStartEvt.tid,
+          frameId,
+        };
+      }
+    }
```

Comment:
can you add a test?

(it seems like something is messed up if this function has test coverage already...I guess a trace was already coming through here that should have been throwing before this PR but is somehow passing whatever test that is both with and without a `NO_TRACING_STARTED` error thrown?)

Reply:
> it seems like something is messed up if this function has test coverage already

oh, ha, that's codecov showing smoke coverage, which is indeed hitting this case, was (sometimes) failing before, and is newly passing :)

Reply:
Added lots of comments, a test trace file, and an extra sanity check on the first resource request